extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -146,9 +161,19 @@ export class ProjectsController {
 	}
 
 	public async publishProjectCallback(project: Project, settings: IPublishSettings | IGenerateScriptSettings): Promise<mssql.DacFxResult | undefined> {
+		const telemetryProps: Record<string, string> = {};
+		const buildStartTime = new Date().getMilliseconds();
+
 		const dacpacPath = await this.buildProject(project);
 
+		const buildEndTime = new Date().getMilliseconds();
+		telemetryProps.buildDuration = (buildEndTime - buildStartTime).toString();
```

Comment:
Yeah - if you care about correlating the two then this is fine. I just figured you'd really just care about the build time no matter where it was started, what kind of information do you think you gain by seeing build times for publish projects specifically?

I would consider having buildProject return the telemetry data though so you don't have to measure it twice. 

(also you could always add your own correlation ID by creating a GUID that's passed to the build function or anything else you want to correlate. So that's an option too if you wanted to keep things even more compartmentalized`

Reply:
I considered the correlation GUID, but then I'm doing a seek on a month of telemetry to actually match them.

I could modify `buildProject()` to return build time in addition to the dacpac path... but since JS doesn't support multiple returns, I'd have to allocate an object and uppack it up here in which case I don't think I'd save anything perf-wise, though I could be wrong on that front.